include/ginkgo/core/base/range_accessors.hpp

```diff
@@ -205,6 +631,289 @@ class row_major {
 };
 
 
+namespace detail_colmajor {
+
+
+/**
+ * This helper runs from first to last dimension in order to compute the index.
+ * The index is computed like this:
+ * indices: x1, x2, x3, ..., xn
+ * compute(stride, x1, x2, x3, ..., x(n-1), xn) ->
+ *  x1 * stride[0] + x2 * stride[1] + ...
+ *    + x(n-2) * stride[n-3] + x(n-1) + xn * stride[n-2]
+ * Note that swap of the last two strides, making this 'column major'.
+ */
+template <typename ValueType, size_type total_dim, size_type current_iter = 1>
+struct index_helper_s {
```

Comment:
Instead of going for a new helper structure, I would recommend rename the `compute` of the `row_major` helper to `compute_row_major`, and name the `column_major` one `compute_column_major`. That would save quite a bit of code duplication.

Reply:
I'm not sure what you mean. This `index_helper_s` does not do the same thing as `row_major_helper_s`. Instead of adding a prefix like 'colmajor' to everything, I just decided on a `detail_colmajor` namespace.

Reply:
The structure provides the same framework for both row- and column-major (`ValueType`, `total_dim` and `current_iter`, which is counted upwards). That is why I suggested to merge them together and just have a different function name. If you want to keep it totally separate, I can understand.

Reply:
Ah, I think I see what you mean now. Yes, it might be more readable to keep them separate at the cost of a few lines of duplication in the struct definition itself.